src/test/regress/expected/gporca_optimizer.out

```diff
@@ -12676,8 +12676,8 @@ where out.b in (select coalesce(tcorr2.a, 99)
    ->  Gather Motion 3:1  (slice1; segments: 3)  (cost=0.00..431.00 rows=1 width=8)
          ->  Seq Scan on tcorr1  (cost=0.00..431.00 rows=1 width=8)
    SubPlan 1  (slice0)
-     ->  Result  (cost=0.00..1324032.56 rows=1 width=4)
-           ->  Nested Loop Left Join  (cost=0.00..1324032.56 rows=1 width=4)
+     ->  Result  (cost=0.00..1324032.56 rows=2 width=4)
+           ->  Nested Loop Left Join  (cost=0.00..1324032.56 rows=2 width=4)
```

Comment:
Please explain, why getting of 2 as estimate rows count is ok here.
[Here](https://github.com/arenadata/gpdb/pull/250#discussion_r709964492) is my old question for reference.

Reply:
This is a correct behaviour within the framework of the logic of greenplum. This happens beacuase ORCA says that after join we get 1.2 rows, and explain rounds this up, so we get 2 where we should get 1. Before this patch was applied, we divided this number by segments count, got value less than 1, round it to 1 and the number of rows turned out to be correct, but it was by accident. This patch doesn't use segments count, it uses gang size instead, so we get incorrect rows number from join, which is 1.2, and divide it by 1, because the calculation is performed in a single segment, then we get 1.2 and round it to 2. Despite the correct logic of the patch, we get incorretc value because of ORCA and greenplum's rounding. For example, vanilla postgres counts rows number correctly and doesn't round rows number at all, so we don't see this misbehaviour there.

Reply:
Good explanation. Can you add a link to the code line where we ceil 1.2 to 2?

Reply:
We round this up right before the output; here is a line for plain text:
https://github.com/arenadata/gpdb/blob/8c5072312aa8bae20033ca3372242c0631e96c17/src/backend/commands/explain.c#L1776
And here's for all the others formats:
https://github.com/arenadata/gpdb/blob/8c5072312aa8bae20033ca3372242c0631e96c17/src/backend/commands/explain.c#L1782